### PR TITLE
Adapt integration testing to run in RHEL8 and beyond

### PR DESCRIPTION
**Pull request by sgala**
This pull request makes most integration tests run for RHEL8, while keeping all RHEL7 tests running.

### CODE REVIEW

File: testsuite/src/test/groovy/noe/modjk/JkConfiguratorTestIT.groovy [hunk 1]
```diff
@@ -61,7 +61,7 @@ class JkConfiguratorTestIT extends TestAbstract {
   static Collection<Object[]> data() {
     Platform platform = new Platform()
     boolean modJkPlatforms =
-            (platform.isRHEL() && (platform.isX64())) ||
+            (platform.isRHEL7() && (platform.isX64())) ||
```
[the remark this answers is not in the document]
Comment (sgala) [reply testsuite/src/test/groovy/noe/modjk/JkConfiguratorTestIT.groovy:64]: RHEL8 test was not running, and I understood Paul  talked mod_jk not being released for RHEL8 and beyond. I might have misunderstood.

**Comment by sgala** [comment]
> There are multiple forms of comments showing which RHEL versions are not supported. Can we unify this?

I don't understand what do you mean, can you be more specific? In general I try to stick to the style of the code I'm fixing, and minimize the changes, but I can be more aggressive if I know what is required.

File: testsuite/src/test/groovy/noe/tomcat/BindingsTomcat7ConfiguratorIT.groovy [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
+import noe.common.utils.Platform
+import org.junit.Assume
 import org.junit.BeforeClass
 
 
 class BindingsTomcat7ConfiguratorIT extends BindingsTomcatConfiguratorIT {
 
   @BeforeClass
   public static void beforeClass() {
+    Assume.assumeTrue("Tomcat 7 not release for >RHEL7", new Platform().isRHEL7())
```
Comment (AdamKrajcik) [inline testsuite/src/test/groovy/noe/tomcat/BindingsTomcat7ConfiguratorIT.groovy:12]: Example of multiple types or limitations. In other places you have RHEL8+

File: testsuite/src/test/groovy/noe/tomcat/JvmRouteTomcat8ConfiguratorIT.groovy [hunk 3]
```diff
@@ -11,7 +11,7 @@ class JvmRouteTomcat8ConfiguratorIT extends JvmRouteTomcatConfiguratorIT {
   public static void beforeClass() {
     Platform platform = new Platform()
     Assume.assumeFalse("JWS is not supported on HP-UX => skipping", platform.isHP())
-    Assume.assumeFalse("JWS 3.0 is not supported on RHEL5 => skipping", platform.isRHEL5())
+    Assume.assumeTrue("JWS 3.0 is not supported on RHEL5 or RHEL8+ => skipping", platform.isRHEL6() || platform.isRHEL7())
```
[the remark this answers is not in the document]
Comment (AdamKrajcik) [reply testsuite/src/test/groovy/noe/tomcat/JvmRouteTomcat8ConfiguratorIT.groovy:14]: Please do.

File: testsuite/src/test/resources/tomcat9-common-test.properties [hunk 4]
```diff
@@ -1,4 +1,4 @@
-ews.version=5.0.0
+ews.version=5.5.0
```
[the remark this answers is not in the document]
Comment (PaulLodge) [reply testsuite/src/test/resources/tomcat9-common-test.properties]: Please can we have consistent versions for all the tests, I propose the latest major releases for JWS and JBCS, 5.7.0 and 2.4.51 respectively.

File: testsuite/src/test/groovy/noe/tomcat/BindingsTomcat7ConfiguratorIT.groovy (hunk 2 above)
Comment (sgala) [reply testsuite/src/test/groovy/noe/tomcat/BindingsTomcat7ConfiguratorIT.groovy:12]: Ah, I see! I'll change it

File: testsuite/src/test/groovy/noe/modjk/JkConfiguratorTestIT.groovy (hunk 1 above)
[the remark this answers is not in the document]
Comment (sgala) [reply testsuite/src/test/groovy/noe/modjk/JkConfiguratorTestIT.groovy]: undoing this change makes [3 new failures](https://webservers-jenkins-csb-jwsqe.apps.ocp-c1.prod.psi.redhat.com/job/noe/job/noe-core/job/noe-core-integration-linux-clone-sgala/34/) appear for RHEL8. They might or might not be related with the remaining  JkConfiguratorTestIT failures for windows, which I was not able to debug.

File: pom.xml [hunk 5]
```diff
@@ -5,6 +5,13 @@
     <groupId>org.jboss.qa.noe</groupId>
     <artifactId>noe-core-parent</artifactId>
     <version>0.17.11-SNAPSHOT</version>
+    <dependencies>
```
Comment (AdamKrajcik) [inline pom.xml:8]: This dependency is in an unusual place.

File: testsuite/src/test/groovy/noe/tomcat/BindingsTomcat8ConfiguratorIT.groovy [hunk 6]
```diff
@@ -11,7 +11,7 @@ class BindingsTomcat8ConfiguratorIT extends BindingsTomcatConfiguratorIT {
   public static void beforeClass() {
     Platform platform = new Platform()
     Assume.assumeFalse("JWS is not supported on HP-UX => skipping", platform.isHP())
-    Assume.assumeFalse("JWS 3.0 is not supported on RHEL5 => skipping", platform.isRHEL5())
+    Assume.assumeTrue("JWS 3.0 is not supported on RHEL5 or >RHEL7 => skipping", platform.isRHEL6() || platform.isRHEL7())
```
Comment (AdamKrajcik) [inline testsuite/src/test/groovy/noe/tomcat/BindingsTomcat8ConfiguratorIT.groovy:14]: Still multiple versions of RHEL limitations. Seen RHEL7+ in another change.

File: testsuite/src/test/groovy/noe/tomcat/JmxTomcat7ConfiguratorIT.groovy [hunk 7]
```diff
@@ -1,12 +1,15 @@
...
+import org.junit.Assume
 import org.junit.BeforeClass
+import noe.common.utils.Platform
 
 
 class JmxTomcat7ConfiguratorIT extends JmxTomcatConfiguratorIT {
 
   @BeforeClass
   public static void beforeClass() {
+    Assume.assumeTrue("Tomcat 7 not release for >RHEL7", new Platform().isRHEL7())
```
Comment (AdamKrajcik) [inline testsuite/src/test/groovy/noe/tomcat/JmxTomcat7ConfiguratorIT.groovy:12]: Still multiple versions of RHEL limitations. Seen RHEL7+ in another change.

File: testsuite/src/test/groovy/noe/tomcat/JmxTomcat8ConfiguratorIT.groovy [hunk 8]
```diff
@@ -11,7 +11,7 @@ class JmxTomcat8ConfiguratorIT extends JmxTomcatConfiguratorIT {
   public static void beforeClass() {
     Platform platform = new Platform()
     Assume.assumeFalse("JWS is not supported on HP-UX => skipping", platform.isHP())
-    Assume.assumeFalse("JWS 3.0 is not supported on RHEL5 => skipping", platform.isRHEL5())
+    Assume.assumeTrue("JWS 3.0 is not supported from RHEL8 onwards => skipping", platform.isRHEL7())
```
Comment (AdamKrajcik) [inline testsuite/src/test/groovy/noe/tomcat/JmxTomcat8ConfiguratorIT.groovy:14]: Still multiple versions of RHEL limitations. Seen RHEL7+ in another change.

File: testsuite/src/test/groovy/noe/tomcat/JvmRouteTomcat7ConfiguratorIT.groovy [hunk 9]
```diff
@@ -1,12 +1,15 @@
...
+import noe.common.utils.Platform
+import org.junit.Assume
 import org.junit.BeforeClass
 
 
 class JvmRouteTomcat7ConfiguratorIT extends JvmRouteTomcatConfiguratorIT {
 
   @BeforeClass
   public static void beforeClass() {
+    Assume.assumeTrue("Tomcat 7 not release for >RHEL7", new Platform().isRHEL7())
```
Comment (AdamKrajcik) [inline testsuite/src/test/groovy/noe/tomcat/JvmRouteTomcat7ConfiguratorIT.groovy:12]: Still multiple versions of RHEL limitations. Seen RHEL7+ in another change.

Comment (sgala) [reply testsuite/src/test/groovy/noe/tomcat/JvmRouteTomcat7ConfiguratorIT.groovy]: Sorry, when I upgraded I only checked the ones you highlighted. Now I have searched the whole diff and edited to consistent wording (I hope)

File: pom.xml (hunk 5 above)
Comment (PaulLodge) [reply pom.xml:8]: Please move the dependency to the correct place below

File: testsuite/src/test/groovy/noe/modjk/JkConfiguratorTestIT.groovy [hunk 10]
```diff
@@ -526,11 +526,20 @@ class JkConfiguratorTestIT extends TestAbstract {
         def sslConfFile = new File(facingServer.getConfDeploymentPath(), "ssl.conf")
         log.debug("Deleting ssl.conf: $sslConfFile.absolutePath")
         JBFile.delete(sslConfFile)
+        // we are not testing cgid here, lets remove cgi.conf (requires cgid with worker/event)
+        def cgiConfFile = new File(facingServer.getConfModulesDeploymentPath(), "01-cgi.conf")
+        log.debug("Deleting cgi.conf: $cgiConfFile.absolutePath")
+        JBFile.delete(cgiConfFile)
```
Comment (PaulLodge) [inline testsuite/src/test/groovy/noe/modjk/JkConfiguratorTestIT.groovy:532]: it might make more sense to use fileVault here instead of deleting it.

Comment (sgala) [reply testsuite/src/test/groovy/noe/modjk/JkConfiguratorTestIT.groovy:532]: This got copied from the lines directly above them (deleting ssl.conf) so I guessed it was safe. If we are refactoring we should refactor both

Comment (PaulLodge) [reply testsuite/src/test/groovy/noe/modjk/JkConfiguratorTestIT.groovy:532]: agreed, please put this on your todo list, ready for the next release

**Pull request merged**